change config Path for more overview if you have a lot plugins installed
Update: place vonp.conf into vomp plugin subdir also

Petric Frank <pfrank@gmx.de> 28 Dec 2007

based on vdr-vompsever-0.2.6-gentoo.diff from Joerg Bornkessel <hd_brummy@gentoo.org> 17 Feb 2007

diff -Naur vompserver-0.2.7.old/bootpd.c vompserver-0.2.7/bootpd.c
--- vompserver-0.2.7.old/bootpd.c	2007-11-05 22:05:30.000000000 +0100
+++ vompserver-0.2.7/bootpd.c	2007-12-28 01:54:44.000000000 +0100
@@ -99,7 +99,7 @@
 
   // Open a config file for the given MAC
 
-  const char* configDir = cPlugin::ConfigDirectory();
+  const char* configDir = cPlugin::ConfigDirectory( "vomp" );
 //  char* configDir = "/opt/dvb/vdr-config-1.3/plugins";
   if (!configDir)
   {
diff -Naur vompserver-0.2.7.old/vompserver.c vompserver-0.2.7/vompserver.c
--- vompserver-0.2.7.old/vompserver.c	2007-11-07 21:28:28.000000000 +0100
+++ vompserver-0.2.7/vompserver.c	2007-12-28 02:15:22.000000000 +0100
@@ -27,6 +27,8 @@
 static const char *VERSION        = "0.2.7";
 static const char *DESCRIPTION    = "VDR on MVP plugin by Chris Tallon";
 
+static char *PLUGIN_CONFIG_SUBDIR = "vomp";
+
 class cPluginVompserver : public cPlugin
 {
 public:
@@ -61,7 +63,7 @@
 bool cPluginVompserver::Start(void)
 {
   // Start any background activities the plugin shall perform.
-  int success = mvpserver.run(configDir);
+  int success = mvpserver.run(configDir ? configDir : PLUGIN_CONFIG_SUBDIR);
   if (success) return true;
   else return false;
 }

